File: bmlsj/03/[PGS - LV2] 모음사전.java
class Solution {
    
    static int answer = 0, cnt = 0;
    static String[] arr = {"A", "E", "I", "O", "U"};
    public int solution(String word) {
        
        dfs("", word);
        return answer;
    }
    
    static void dfs(String curr, String target) {
        
        if (curr.equals(target)) {
            answer = cnt;
            System.out.println(cnt);
            return;
        }
        
        if (curr.length() == 5) return;
        
        for (int i = 0; i < 5; i++) {
            cnt++;
            dfs(curr + arr[i], target);
        }
    }
}

File: hyuksanglee/02/[PGS - Lv2] 모음 사전.js
const alpa = ['A','E','I','O','U'];

function solution(word) {
    var answer = 0;
    let count = 0;
    for(const c of alpa){
        let total ="";
        const result = dfs(word, total+c, count+1);
        count = result[1];
        
        // 💡 변수 count 대신 result[1]을 직접 리턴하셔야 안전합니다!
        if(result[0]) return result[1]; 
        
    }
    return answer;
}

function dfs(word, total, count){
    if(word===total){
        return [true,count];
    }
    if(total.length===5){
        return [false,count];
    }
    for(const w of alpa){
        const result = dfs(word,total+w, count+1);
        count = result[1];
        if(result[0]) return [true,count];
    }
    return [false,count];
}
